core/common/test/DateTimePeriodTest.kt

```diff
@@ -12,6 +12,39 @@ import kotlin.time.*
 
 class DateTimePeriodTest {
 
+    @Test
+    fun normalization() {
+        assertPeriodComponents(DateTimePeriod(years = 1) as DatePeriod, years = 1)
```

Comment:
Can you use this assertion function in tests for `parse` as well?

Reply:
Sure, but, in my opinion, it would make those tests less clear. For example,
```kotlin
         assertEquals(DateTimePeriod(months = 14), DateTimePeriod.parse("P14M"))
```
clearly implies that `DateTimePeriod(months = m) == DateTimePeriod.parse("P${m}M")`, which is a property that we want. Seeing the result in terms of values of class properties would obscure this. Moreover, the values class properties are now being tested with `normalization()`.

What benefits do you see in using this function for tests of parsing?

Reply:
By mixing parsing and normalized equality this test makes an impression that the resulting properties of the parsed `DateTimePeriod` would be the same as those passed to the constructor-like function of expected value. This is misleading.

Reply:
For some unfathomable reason, I didn't see a notification about your response. I am sorry for the delay.

I still disagree and consider these concerns orthogonal: we want parsing to behave the same way construction does, and we want construction to produce specific values of properties. Parsing is not semantically connected to the values of properties, and mixing these rubs me the wrong way. However, I don't consider this issue important enough to die on this hill, so I added a commit to accommodate your preferences, though I would be glad to revert it were you to change your mind about this.

Reply:
As a compromise, we could leave the equality assertions in the parsing test, but add `assertPeriodComponents` for those values that cause normalization to happen. This would be done mostly for explanatory reasons.

Reply:
I added checks for actual values of properties to places where normalization does affect them.